internal/state/relationship/capturer.go

```diff
@@ -63,32 +84,89 @@ func (c *CapturerImpl) Capture(obj client.Object) {
 				Name:      svcName,
 			}
 		}
+	case *v1beta1.Gateway:
+		var selectors []labels.Selector
+		for _, listener := range o.Spec.Listeners {
+			if selector := graph.GetAllowedRouteLabelSelector(listener); selector != nil {
+				convertedSelector, err := metav1.LabelSelectorAsSelector(selector)
+				if err == nil {
+					selectors = append(selectors, convertedSelector)
+				}
+			}
+		}
+
+		nsname := client.ObjectKeyFromObject(o)
+		if len(selectors) > 0 {
```

Comment:
what if labels get updated and some namespace is no longer selected. will we need to update the corresponding element in c.referencedNamespace?

Reply:
Yeah, I think I missed that case.